qabelbox/src/main/java/de/qabel/qabelbox/storage/StorageSearch.java

```diff
@@ -340,23 +320,11 @@ private void addAll(List<BoxObject> lst) throws QblStorageException {
 
     @Override
     public StorageSearch clone() throws CloneNotSupportedException {
-        return new StorageSearch(cloneResultList(getResults()), (Hashtable<String, BoxObject>) getPathMapping().clone());
-    }
-
-    private List<BoxObject> cloneResultList(List<BoxObject> list) throws CloneNotSupportedException {
-        List<BoxObject> clone = new ArrayList<>(list.size());
-        for (BoxObject item : list) {
-            if (item instanceof BoxFile) {
-                clone.add(((BoxFile) item).clone());
-            }
-            if (item instanceof BoxExternalFile) {
-                clone.add(((BoxExternalFile) item).clone());
-            } else if (item instanceof BoxExternalFolder) {
-                clone.add(((BoxExternalFolder) item).clone());
-            } else if (item instanceof BoxFolder) {
-                clone.add(((BoxFolder) item).clone());
-            }
+        try {
+            return new StorageSearch(navigation, new ArrayList<>(results));
+        } catch (QblStorageException e) {
+            throw new CloneNotSupportedException();
```

Comment:
It is weird that sometimes cloning is support and sometimes not. The Exception should not be swallowed.